homeassistant/components/mqtt/vacuum/schema_legacy.py

```diff
@@ -236,73 +275,61 @@ def _setup_from_config(self, config):
                 CONF_ERROR_TEMPLATE,
                 CONF_FAN_SPEED_TEMPLATE,
             )
+            if key in config
         }
 
-    def _prepare_subscribe_topics(self):
+    def _prepare_subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
-        for tpl in self._templates.values():
-            if tpl is not None:
-                tpl = MqttValueTemplate(tpl, entity=self)
 
         @callback
         @log_messages(self.hass, self.entity_id)
         def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT message."""
             if (
                 msg.topic == self._state_topics[CONF_BATTERY_LEVEL_TOPIC]
-                and self._templates[CONF_BATTERY_LEVEL_TEMPLATE]
+                and CONF_BATTERY_LEVEL_TEMPLATE in self._config
             ):
-                battery_level = self._templates[
+                battery_level: ReceivePayloadType = self._templates[
                     CONF_BATTERY_LEVEL_TEMPLATE
-                ].async_render_with_possible_json_value(
-                    msg.payload, PayloadSentinel.DEFAULT
-                )
+                ](msg.payload, PayloadSentinel.DEFAULT)
                 if battery_level and battery_level is not PayloadSentinel.DEFAULT:
                     self._attr_battery_level = max(0, min(100, int(battery_level)))
 
             if (
                 msg.topic == self._state_topics[CONF_CHARGING_TOPIC]
-                and self._templates[CONF_CHARGING_TEMPLATE]
+                and CONF_CHARGING_TEMPLATE in self._config
```

Comment:
I find this confusing. Although correct it would be easier to understand with
```suggestion
                and CONF_CHARGING_TEMPLATE in self._templates
```